### DISCUSSION TITLE
AssemblyRunner Start method

### DISCUSSION THREAD

**Discussion by hel1e** [xunit/xunit#2933] [General]
The introduction of a new override of the method Start caused a confusing situation. 

This made our call - without any arguments - ambiguous. 
So we added a long list of nulls to make the code compile. But then it failed with the following error message:

```
Unhandled exception. System.ArgumentException: Argument was empty (Parameter 'typeName') at
 Guard.ArgumentNotNullOrEmpty(String argName, IEnumerable argValue) in /_/src/common/Guard.cs:line 27
```

So even if the default value of typeName is null, actively passing null causes the method to fail.

This is in a .net 8.0 project, where nullable is disabled by default. So to make it work, we added the following to the project file:

```
	<PropertyGroup>
		<Nullable>enable</Nullable>
	</PropertyGroup>
```

So now - even if the code does look a bit ugly - it finally works.

**Comment by bradwilson** [comment]
Apologies for that.

There is a change already coming to 2.8.1 to address this, and it involves switching to a new `AssemblyRunnerStartOptions` structure: https://github.com/xunit/xunit/commit/ed625e5ec35dabae1d485d22e107203f112a7786

This should allow further expansion without undergoing this kind of signature issue in the future.

You can test it today with any CI version starting with `2.8.1-pre.4`. I am planning to ship a final release for 2.8.1 sometime in the next few days. https://xunit.net/docs/using-ci-builds
